src/components/Checkbox/Checkbox.st.css

```diff
@@ -95,7 +109,44 @@
     color: value(CurrentTextColor);
 }
 
+.suffix {
+    margin-left: 5px;
+}
+
+.box .suffix {
+    margin-left: auto;
+}
+
 /* STATES */
+.box.disabled {
+    cursor: default;
+}
+
+.box:hover, .box:focus-within {
+    border-color: applyOpacity(value(CurrentBorderColor), 1);
+}
+
+.box.disabled:hover:not(.checked), .box.disabled:focus-within:not(.checked) {
+    border-color: applyOpacity(value(CurrentBorderColor), 0.6);
+}
+
+.box.checked {
+    background-color: applyOpacity(value(CurrentIconColor), 0.1);
+    border-color: value(CurrentIconColor);
+}
+
+.box.checked:not(.disabled):hover, .box.checked:not(.disabled):focus-within {
+    border-color: applyOpacity(value(CurrentIconColor), 0.6);
+}
+
+.box:not(.disabled):hover .icon {
+    border-color: applyOpacity(value(CurrentBorderColor), 1);
+}
+
+.box:last-of-type {
+    margin-bottom: 0;
+}
```

Comment:
It seems this code should be in the CheckboxGroup component's stylesheet.
We usually don't want the component itself affecting its surrounding.